### PR TITLE
Use team_size_max to fix "Team size too large" error in reducer test

### LINKED ISSUE #6699: Nightly unit test failure, OpenMP backend, gcc/8.3.0: openmp.reducers_int Kokkos::abort: Requested Team Size is too large!
**Describe the bug**

Nightly tests with gcc/8.3.0 and the OpenMP backend failed in the `openmp.reducers_int` subtest (both debug and rel builds):
```
[ RUN      ] openmp.reducers_int
Kokkos::abort: Requested Team Size is too large!
Backtrace:
[0x1103d1c4] 
[0x1100e268] 
[0x105de050] 
[0x105df410] 
[0x105d1f9c] 
[0x1100c30c] 
[0x10ff9370] 
[0x10ff9830] 
[0x10ffa148] 
[0x10ffce34] 
[0x10ffd4d0] 
[0x10012508] 
[0x200002599f5c] 
[0x20000259a0f4] __libc_start_main
```

Test failure began after merge of the following changes:

**Changes**

    Fixup test math functions ulp should double -> int ([details](https://jenkins-son.sandia.gov/job/Kokkos_develop_weaver_cuda_openmp_cuda1122_gcc830_volta/488/changes#71729af719b2eb20b512efad05ae609e74e1dd7a) / [githubweb](https://github.com/kokkos/kokkos/commit/71729af719b2eb20b512efad05ae609e74e1dd7a))
    #5333: CUDA: Use scratch space appropriate to small reduction elements in Team reductions (#5334) ([details](https://jenkins-son.sandia.gov/job/Kokkos_develop_weaver_cuda_openmp_cuda1122_gcc830_volta/488/changes#02b46c09c75697da5a1e4bfed5429a56c1bfb7ef) / [githubweb](https://github.com/kokkos/kokkos/commit/02b46c09c75697da5a1e4bfed5429a56c1bfb7ef))
    Cuda: Allocate using the correct device (#6392) ([details](https://jenkins-son.sandia.gov/job/Kokkos_develop_weaver_cuda_openmp_cuda1122_gcc830_volta/488/changes#4078a0d8a022ab1ad2291554b2820916fa3bc361) / [githubweb](https://github.com/kokkos/kokkos/commit/4078a0d8a022ab1ad2291554b2820916fa3bc361))
    Let `Profiling::ProfilingSection(std::string)` constructor be explicit and nodiscard (#6690) ([details](https://jenkins-son.sandia.gov/job/Kokkos_develop_weaver_cuda_openmp_cuda1122_gcc830_volta/488/changes#f485cfa53cb2d25cab98e7a1d41a116027ec2363) / [githubweb](https://github.com/kokkos/kokkos/commit/f485cfa53cb2d25cab98e7a1d41a116027ec2363))

**Please include the following for a minimal reproducer**
**Reproducer** (weaver rhel8 queue):
```bash
source /projects/ppc64le-pwr9-rhel8/legacy-env.sh
module load cuda/11.2.2/gcc/8.3.1 cmake/3.23.1

${KOKKOS_PATH}/generate_makefile.bash --with-cuda --with-openmp --compiler=${KOKKOS_PATH}/bin/nvcc_wrapper --arch=Volta70,Power9 --with-cuda-options=enable_lambda --kokkos-path=${KOKKOS_PATH} ${DEBUG} --cxxflags=${CXXFLAGS} --cxxstandard=17

make -j16

```

### PR DESCRIPTION
**Pull request by tcclevenger**
A recently added reducer test was using a teamsize that was too large in some configurations. Using `TeamPolicy::team_size_max()` as a upper bound to ensure team size is not too large.

I also reorganized the test to use a functor and changed the test with league size 10 to store results in a view size 10 since previously the result was stored in a dim-0 view and wa just being overwritten by the last team to store it. Maybe the test was supposed to have league size 1?

Closes https://github.com/kokkos/kokkos/issues/6699

### CODE REVIEW

File: core/unit_test/TestReducers.hpp [hunk 1]
```diff
@@ -368,36 +389,39 @@ struct TestReducers {
...
+      Kokkos::View<Scalar*, ExecSpace> result("result", league_size);
+      TeamSumNestedFunctor tnf(f, league_size, N, result);
+      auto team_size_max = Kokkos::TeamPolicy<ExecSpace>(league_size, 1)
+                               .team_size_max(tnf, Kokkos::ParallelForTag());
+      auto team_size = std::min(team_size_max, TEST_EXECSPACE().concurrency());
+      Kokkos::parallel_for(
+          Kokkos::TeamPolicy<ExecSpace>(league_size, team_size), tnf);
+      auto result_h =
+          Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace(), result);
+      for (auto i = 0; i < result_h.extent(0); ++i) {
```
Comment (dalg24) [inline core/unit_test/TestReducers.hpp:421]:
```suggestion
      for (int i = 0; i < result_h.extent_int(0); ++i) {
```
```
/__w/kokkos/kokkos/core/unit_test/TestReducers.hpp:421:26: error: comparison of integer expressions of different signedness: 'int' and 'std::enable_if_t<true, long unsigned int>' {aka 'long unsigned int'} [-Werror=sign-compare]
  421 |       for (auto i = 0; i < result_h.extent(0); ++i) {
      |                        ~~^~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors

**Review by crtrott — APPROVED** [review]

**Review by ndellingwood — APPROVED** [review]
I confirmed the test now passes in the previously failing build, thanks @tcclevenger !

**Comment by ndellingwood** [comment]
OpenACC CI check failed in the Kokkos_CoreUnitTest_OpenACC test:

```
4: [ RUN      ] openacc.view_allocation_error
4: Kokkos::RangePolicy bounds error: The lower bound (0) is greater than the upper bound (-43).
4: Backtrace:
4: [0x148f7d3] 
4: [0x1444563] 
4: [0x44bbe3] 
4: [0x44d1a5] 
4: [0x450d07] 
4: [0x44ca20] 
4: [0x57a305] 
4: [0x56f295] 
4: [0x95a41d] 
4: [0x1437303] 
4: [0x140094f] 
4: [0x1402c9f] 
4: [0x1403fc7] 
4: [0x1416ae7] 
4: [0x143a143] 
4: [0x140538e] 
4: [0x4243b5] 
4: [0x7f3b32e62083] __libc_start_main
4: [0x4242ae] 
```

Unrelated to this PR, which does not alter the `view_allocation_error` test

**Comment by dalg24** [comment]
This test is just bad.  It does not fail if no exception is thrown.
My best guess is it does not throw and we proceed to the view initialization which hard codes `int64_t` as the index type for the range policy.

**Comment by seyonglee** [comment]
> This test is just bad. It does not fail if no exception is thrown. My best guess is it does not throw and we proceed to the view initialization which hard codes `int64_t` as the index type for the range policy.

How about skipping the TestViewAPI<>::run_test_error() for the OpenACC backend, as the OpenMPTarget backend does?

**Comment by dalg24** [comment]
> > This test is just bad. It does not fail if no exception is thrown. My best guess is it does not throw and we proceed to the view initialization which hard codes `int64_t` as the index type for the range policy.
> 
> How about skipping the TestViewAPI<>::run_test_error() for the OpenACC backend, as the OpenMPTarget backend does?

See #6729

**Review by masterleinad — APPROVED** [review]

**Pull request merged**